app/main.py

```diff
@@ -1,213 +1,51 @@
-
-    # apply armour
-    mordred["protection"] = 0
-    for a in mordred["armour"]:
-        mordred["protection"] += a["protection"]
-
-    # apply weapon
-    mordred["power"] += mordred["weapon"]["power"]
-
-    # apply potion if exist
-    if mordred["potion"] is not None:
-        if "power" in mordred["potion"]["effect"]:
-            mordred["power"] += mordred["potion"]["effect"]["power"]
-
-        if "protection" in mordred["potion"]["effect"]:
-            mordred["protection"] += mordred["potion"]["effect"]["protection"]
-
-        if "hp" in mordred["potion"]["effect"]:
-            mordred["hp"] += mordred["potion"]["effect"]["hp"]
-
-    # red_knight
-    red_knight = knightsConfig["red_knight"]
-
-    # apply armour
-    red_knight["protection"] = 0
-    for a in red_knight["armour"]:
-        red_knight["protection"] += a["protection"]
+from app.knights_config import KNIGHTS
+from app.knight_crearing import create_knight
 
-    # apply weapon
-    red_knight["power"] += red_knight["weapon"]["power"]
 
-    # apply potion if exist
-    if red_knight["potion"] is not None:
-        if "power" in red_knight["potion"]["effect"]:
-            red_knight["power"] += red_knight["potion"]["effect"]["power"]
+def battle(knights: dict) -> dict:
+    lancelot = create_knight(knights["lancelot"])
+    lancelot.apply_armour(knights["lancelot"])
+    lancelot.apply_weapon(knights["lancelot"])
+    lancelot.apply_potion(knights["lancelot"]["potion"])
 
-        if "protection" in red_knight["potion"]["effect"]:
-            red_knight["protection"] += red_knight["potion"]["effect"]["protection"]
+    arthur = create_knight(knights["arthur"])
+    arthur.apply_armour(knights["arthur"])
+    arthur.apply_weapon(knights["arthur"])
+    arthur.apply_potion(knights["arthur"]["potion"])
 
-        if "hp" in red_knight["potion"]["effect"]:
-            red_knight["hp"] += red_knight["potion"]["effect"]["hp"]
+    mordred = create_knight(knights["mordred"])
+    mordred.apply_armour(knights["mordred"])
+    mordred.apply_weapon(knights["mordred"])
+    mordred.apply_potion(knights["mordred"]["potion"])
 
-    # -------------------------------------------------------------------------------
-    # BATTLE:
+    red_knight = create_knight(knights["red_knight"])
+    red_knight.apply_armour(knights["red_knight"])
+    red_knight.apply_weapon(knights["red_knight"])
+    red_knight.apply_potion(knights["red_knight"]["potion"])
 
-    # 1 Lancelot vs Mordred:
-    lancelot["hp"] -= mordred["power"] - lancelot["protection"]
-    mordred["hp"] -= lancelot["power"] - mordred["protection"]
+    lancelot.hp -= mordred.power - lancelot.protection
+    mordred.hp -= lancelot.power - mordred.protection
 
-    # check if someone fell in battle
-    if lancelot["hp"] <= 0:
-        lancelot["hp"] = 0
+    if lancelot.hp <= 0:
+        lancelot.hp = 0
 
-    if mordred["hp"] <= 0:
-        mordred["hp"] = 0
+    if mordred.hp <= 0:
+        mordred.hp = 0
```

Comment:
Check the checklist 
![image](https://user-images.githubusercontent.com/104218550/205488603-baadcc31-152f-4486-9d1e-758f33f2b1a2.png)